tool/tsh/common/kube_proxy_test.go

```diff
@@ -86,6 +89,176 @@ func (p *kubeTestPack) testProxyKube(t *testing.T) {
 	})
 }
 
+func TestProxyKubeComplexSelectors(t *testing.T) {
+	testenv.WithInsecureDevMode(t, true)
+	testenv.WithResyncInterval(t, 0)
+	kubeFoo := "foo"
+	kubeFooBar := "foo-bar"
+	kubeBaz := "baz"
+	kubeFooLeaf := "foo"
+	ctx := context.Background()
+	s := newTestSuite(t,
+		withRootConfigFunc(func(cfg *servicecfg.Config) {
+			cfg.Auth.NetworkingConfig.SetProxyListenerMode(types.ProxyListenerMode_Multiplex)
+			cfg.Kube.Enabled = true
+			cfg.Kube.ListenAddr = utils.MustParseAddr(localListenerAddr())
+			cfg.Kube.KubeconfigPath = newKubeConfigFile(t, kubeFoo, kubeFooBar, kubeBaz)
+			cfg.Kube.StaticLabels = map[string]string{"env": "root"}
+		}),
+		withLeafCluster(),
+		withLeafConfigFunc(
+			func(cfg *servicecfg.Config) {
+				cfg.Auth.NetworkingConfig.SetProxyListenerMode(types.ProxyListenerMode_Multiplex)
+				cfg.Kube.Enabled = true
+				cfg.Kube.ListenAddr = utils.MustParseAddr(localListenerAddr())
+				cfg.Kube.KubeconfigPath = newKubeConfigFile(t, kubeFooLeaf)
+				cfg.Kube.StaticLabels = map[string]string{"env": "leaf"}
+			},
+		),
+		withValidationFunc(func(s *suite) bool {
+			rootClusters, err := s.root.GetAuthServer().GetKubernetesServers(ctx)
+			require.NoError(t, err)
+			leafClusters, err := s.leaf.GetAuthServer().GetKubernetesServers(ctx)
+			require.NoError(t, err)
+			return len(rootClusters) == 3 && len(leafClusters) == 1
+		}),
+	)
+	rootClusterName := s.root.Config.Auth.ClusterName.GetClusterName()
+	leafClusterName := s.leaf.Config.Auth.ClusterName.GetClusterName()
+	mustLoginSetEnv(t, s)
+	// Set default kubeconfig to a non-exist file to avoid loading other things.
+	t.Setenv("KUBECONFIG", path.Join(os.Getenv(types.HomeEnvVar), uuid.NewString()))
```

Comment:
```suggestion
	t.Setenv(teleport.KubeConfigFile, path.Join(os.Getenv(types.HomeEnvVar), uuid.NewString()))
```

Reply:
teleport.EnvKubeConfig*

changed to use the constant, thank you